src/Symfony/Component/Config/Definition/Builder/ArrayNodeDefinition.php

```diff
@@ -372,7 +376,7 @@ protected function createNode()
                 $node->setKeyAttribute($this->key, $this->removeKeyItem);
             }
 
-            if (true === $this->atLeastOne) {
+            if (true === $this->atLeastOne || false === $this->allowEmptyValue) {
```

Comment:
Fine either way... the numeric node does that actually.

Personally i'd favor `Array::cannotBeEmpty` over `requiresAtLeastOneElement`, but maybe that's just me :)

Maybe provide `requiresAtLeastOneElement` as an alias to `cannotBeEmpty`?